From d87435a0aeb62db693a063af798fc1d86f7a2b09 Mon Sep 17 00:00:00 2001
From: Marcel Bargull <marcel.bargull@udo.edu>
Date: Fri, 22 Dec 2023 15:46:57 +0100
Subject: [PATCH 2/2] Disable direct invocation of linker if LD is set

It's apparent that lddlflags is just as ldflags something to be used
when invoking the linker through the compiler.

Removing the direct usage of the linker if the LD environment variable
is set (by our compiler/linker activation packages) fixes
https://github.com/conda-forge/perl-feedstock/issues/59#issuecomment-1623967103
.

Signed-off-by: Marcel Bargull <marcel.bargull@udo.edu>
---
 dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm b/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm
index 0494cafd95..f2c1935e03 100644
--- a/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm
+++ b/dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Base.pm
@@ -41,7 +41,7 @@ sub new {
      if defined $ENV{CFLAGS};
   $self->{config}{cxx} = $ENV{CXX} if defined $ENV{CXX};
   $self->{config}{cxxflags} = $ENV{CXXFLAGS} if defined $ENV{CXXFLAGS};
-  $self->{config}{ld} = $ENV{LD} if defined $ENV{LD};
+  # $self->{config}{ld} = $ENV{LD} if defined $ENV{LD};
   $self->{config}{ldflags} = join(" ", $self->{config}{ldflags}, $ENV{LDFLAGS})
      if defined $ENV{LDFLAGS};
 
